CMakeLists.txt

```diff
@@ -278,7 +278,7 @@ if (CMAKE_CROSSCOMPILING)
   search_openblas(0.3.13)
 endif()
 
-if (DISABLE_DOWNLOADS)
+if (DOWNLOAD_DEPENDENCIES)
```

Comment:
I'm trying to understand why linking on Windows is failing, but the output doesn't provide much information.  Do you think you can add something like
```
message(STATUS "Armadillo libraries: ${ARMADILLO_LIBRARIES}")
```
down where `MLPACK_LIBRARIES` is set?  That might provide more of a clue of what is going on. :+1:

I dug deeper in the log too to find the linking message:
```
 C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30037\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE /OUT:"D:\a\1\s\build\Release\mlpack_approx_kfn.exe" /INCREMENTAL:NO /NOLOGO "-static" ..\..\..\..\Release\mlpack.lib C:\hostedtoolcache\windows\OpenBLAS.0.2.14.1\lib\native\lib\x64\libopenblas.dll.a C:\hostedtoolcache\windows\OpenBLAS.0.2.14.1\lib\native\lib\x64\libopenblas.dll.a kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /PDB:"D:/a/1/s/build/Release/mlpack_approx_kfn.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"D:/a/1/s/build/Release/mlpack_approx_kfn.lib" /MACHINE:X64  /machine:x64 mlpack_approx_kfn.dir\Release\approx_kfn_main.obj
 ```
so, it looks like it is linking against OpenBLAS, but compilation is failing because symbols like `_wrapper_ddot` and so forth are not found.  To me, this indicates that compilation was done with `ARMA_USE_WRAPPER` set to true, but then the linking is done as if there is no Armadillo library.

If you print `ARMADILLO_LIBRARIES`, I think that will show us whether or not the Armadillo library has been found.  I suspect that it won't be found---and then maybe we can fix it by modifying the Azure build instructions to specify the `-DARMADILLO_LIBRARY` option when configuring mlpack.  That might work, but that is kind of just a guess...

Reply:
I will give it a try, I have already spent a couple of hours this week trying to figure out what is happening. I can not see why adding ensmallen result in lost armadillo. The only theory I have is related to armadillo version. It might be magically finding the armadillo headers or library packaged with ensmallen nuget package instead of the version we have already installed.
The only reason made me think of such a theory is related to the configuration when CMake is run, it should print the path of armadillo library, but instead it is  only printing the headers:
```
-- Found Armadillo: D:/a/1/s/armadillo-8.400.0/include (found suitable version "8.400.0", minimum required is "8.400.0")
```
Anyway, I will print `ARMADILLO_LIBRARIES` and see where it will guide us :+1:

Reply:
Yes, I was thinking the same thing.  Another option here might be to forcibly disable the wrapper on Windows by modifying `armadillo_bits/config.hpp` to ensure that `ARMA_USE_WRAPPER` is not set.

Reply:
It seems that it is not setting the Armadillo library to a specific `.dll` file, the variable has only `openblas files`. Please have a look: https://dev.azure.com/mlpack/mlpack/_build/results?buildId=7193&view=logs&j=934eac7e-378e-5e05-ed87-f05e34fa9fdc&t=9715bafe-ea9f-54fa-e80f-ab012eb34e65&l=45